### ISSUE TITLE
Functions that take variadic tuple doesn't recognize empty tuple

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4876]
### Discussed in https://github.com/microsoft/pyright/discussions/4873

<div type='discussions-op-text'>

<sup>Originally posted by **sylee957** March 30, 2023</sup>
Given the `int, ... -> str` functions:

```python3
from typing import Callable


def a(var: tuple[()]) -> str:
    raise NotImplementedError


def b(var: tuple[int, int]) -> str:
    raise NotImplementedError


def c(var: tuple[int, int, int]) -> str:
    raise NotImplementedError
```

I recognize that the empty tuple function does not pass the type check

```
f: Callable[[tuple[int, ...]], str]
f = a # Does not pass type check
f = b
f = c
```

While the empty tuple passes the type checking.
```
x: tuple[int, ...]
x = ()
x = (1,)
x = (1, 2)
```
I'm using pyright 1.1.301</div>

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.302, which I just published. It will also be included in a future release of pylance.
